p2p/chain.py

```diff
@@ -246,13 +249,28 @@ def _should_skip_receipts_download(self, header: BlockHeader) -> bool:
         return header.receipt_root == self.chaindb.empty_root_hash
 
     async def request_receipts(self, headers: List[BlockHeader]) -> None:
-        peer = await self.peer_pool.get_random_peer()
-        cast(ETHPeer, peer).sub_proto.send_get_receipts([header.hash for header in headers])
+        peer = await self.get_idle_peer()
+        peer.sub_proto.send_get_receipts([header.hash for header in headers])
+        self._peers_with_pending_requests.add(peer)
         self.logger.debug("Requesting %d block receipts to %s", len(headers), peer)
         now = time.time()
         for header in headers:
             self._pending_receipts[header.receipt_root] = (header, now)
 
+    async def get_idle_peer(self) -> ETHPeer:
+        """Return a random peer which we're not already expecting a response from."""
+        while True:
+            idle_peers = [
+                peer
+                for peer in self.peer_pool.peers
+                if peer not in self._peers_with_pending_requests
+            ]
+            if idle_peers:
+                return cast(ETHPeer, random.choice(idle_peers))
```

Comment:
This is a good candidate for `secrets.choice`

Reply:
I thought about that too, but couldn't think of any reasons why it'd be preferred over plain `random`. Can you think of any?

Reply:
Yes, the `random` module provides no guarantees about providing good sources of randomness and has explicit warnings at the top of the module that the functions should not be used for "security purposes".  The `secrets` module uses secure sources of randomness by default which I don't believe is the case for `random`.

Reply:
Right, but I don't think we need true randomness here -- this is just a way to pick, arbitrarily, one of our connected peers. I believe where we'd need true randomness is where we chose which peers we connect to.

Reply:
I would *strongly* prefer we use strong sources of randomness.  I know there's not much of an attack vector here, but it's cheap to use the *better* version of `choice`.

Reply:
Fair enough